block_writer/main.go

```diff
@@ -179,38 +188,60 @@ func main() {
 	writers := make([]blockWriter, *concurrency)
 
 	errCh := make(chan error)
+	var wg sync.WaitGroup
 	for i := range writers {
+		wg.Add(1)
 		writers[i] = newBlockWriter(db)
-		go writers[i].run(errCh)
+		go writers[i].run(errCh, &wg)
```

Comment:
We're synchronously adding counts to the wait group at line 193 and then kicking off the goroutine to wait at line 202. I don't see where a race can occur. Am I missing something?